util/saconfig

```diff
@@ -61,74 +62,95 @@ def calibrate(dev, channels):
         logging.info("Calibrating channel %d", channel)
         # set calibration options
         dev.SetProperty_i32(smaract.SA_CTLDLL.SA_CTL_PKEY_CALIBRATION_OPTIONS,
-                        channel, calibration_options)
+                            channel, calibration_options)
 
         dev.Calibrate(channel)
 
 
-def autozero(dev, channels):
+# Reference mode to set
+NORMAL_REF, AUTO_ZERO_REF = 0, 1
+
+
+def set_autozero_reference(dev, channels):
     if not channels:
         return
-
-    # Set auto-zero mode
+    # Set auto-zero referencing mode
     for channel in channels:
         logging.info("Setting auto-zero mode for channel %d", channel)
         dev.SetProperty_i32(smaract.SA_CTLDLL.SA_CTL_PKEY_REFERENCING_OPTIONS,
                             channel, smaract.SA_CTLDLL.SA_CTL_REF_OPT_BIT_AUTO_ZERO)  # auto zero
 
-    # Run the referencing
-    logging.info("Starting reference...")
-    dev.reference().result()
-    time.sleep(0.1)
-    logging.info("Complete")
 
+def set_normal_reference(dev, channels):
+    if not channels:
+        return
     # Set normal referencing mode
-    for channel in  channels:
+    for channel in channels:
         logging.info("Setting the reference mode to normal for channel %d", channel)
         dev.SetProperty_i32(smaract.SA_CTLDLL.SA_CTL_PKEY_REFERENCING_OPTIONS,
                             channel, smaract.SA_CTLDLL.SA_CTL_REF_OPT_BIT_NORMAL)  # normal
 
 
+def reference(dev, channels, mode=NORMAL_REF):
+    # Set referencing mode, currently either normal or autozero
+    if mode == NORMAL_REF:
+        set_normal_reference(dev, channels)
+    elif mode == AUTO_ZERO_REF:
+        set_autozero_reference(dev, channels)
```

Comment:
The more I'm looking at the code, the less it makes sense. 
`dev.reference()` calls `_doReference()`, which... sets the referencing to normal! So setting autozero would have no effect here. I'm really confused, because I'm pretty sure we've run this code, and it did have an effect!

Also, I think that this PKEY_REFERENCING_OPTIONS is normally made to contain other info too, which we might want to retain. That's why it's called OPT_**BIT**_AUTO_ZERO. We should just set/unset this bit (3rd bit). With the constant OPT_BIT_NORMAL = 0, it's confusing. From the developer's doc, it seems bits 1,2,3 can be combined. The value OPT_BIT_NORMAL is just there as a helper constant to reset the behaviour.

So I'd suggest to do three changes.
* Firstly, just have a single function `set_reference_autozero(channel, active: bool)`. It reads the current referencing option,  set/unset the bit SA_CTL_REF_OPT_BIT_AUTO_ZERO (based on active flag), and store back the value.
* Secondly, after a referencing, *always* reset to non-auto-zero mode.
* Thirdly, change the smaract MCS2 driver to not change the referencing options, ever. For extra safety, you could read the option at init, and log them + log a warning if AUTO_ZERO is set.